File: frontend/app/types/mission.ts
// Mirrors backend/app/schemas/mission.py / app/schemas/ai_correlation.py /
// app/ai/schemas.py::ReportProposal field-for-field (Phase 18).
export type MissionStatus = "DRAFT" | "APPROVED" | "RUNNING" | "COMPLETED" | "FAILED" | "CANCELLED";
export type MissionStepStatus = "PENDING" | "SKIPPED" | "QUEUED" | "SUCCESS" | "FAILED" | "CANCELLED";
export type SuggestionStatus = "PENDING" | "ACCEPTED" | "DISMISSED";

export interface MissionStep {
  id: string;
  step_order: number;
  label: string;
  tool: string | null;
  profile: string | null;
  options: Record<string, unknown>;
  rationale: string;
  status: MissionStepStatus;
  job_id: string | null;
  skip_reason: string | null;
  created_at: string;
}

export interface Mission {
  id: string;
  project_id: string | null;
  target_id: string;
  goal: string;
  status: MissionStatus;
  max_steps: number;
  created_at: string;
  approved_at: string | null;
  finished_at: string | null;
  steps: MissionStep[];
}

export interface AICorrelationSuggestion {
  id: string;
  finding_id: string;
  related_finding_id: string;
  rationale: string;
  status: SuggestionStatus;
  created_at: string;
  reviewed_at: string | null;
}

export interface ReportProposal {
  title: string;
  job_ids: string[];
  rationale: string;
}

File: backend/app/schemas/mission.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field

from app.models.mission import MissionStatus, MissionStepStatus


class MissionCreateRequest(BaseModel):
    # Mandatory and always a real, registered Asset -- unlike JobCreateRequest,
    # a Mission never accepts free-text target: every step must be
    # re-checkable against is_executable() at execution time, which requires
    # a real Asset row (see app/ai/orchestrator.py).
    target_id: uuid.UUID
    goal: str
    max_steps: int = Field(default=10, gt=0, le=50)


class MissionStepResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    step_order: int
    label: str
    tool: str | None
    profile: str | None
    options: dict
    rationale: str
    status: MissionStepStatus
    job_id: uuid.UUID | None
    skip_reason: str | None
    created_at: datetime


class MissionResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    project_id: uuid.UUID | None
    target_id: uuid.UUID
    goal: str
    status: MissionStatus
    max_steps: int
    created_at: datetime
    approved_at: datetime | None
    finished_at: datetime | None
    steps: list[MissionStepResponse] = Field(default_factory=list)
